tools/apprunner-gradle-plugin/src/main/java/org/projectnessie/quarkus/gradle/QuarkusAppPlugin.java

```diff
@@ -18,27 +18,58 @@
 import org.gradle.api.Action;
 import org.gradle.api.Plugin;
 import org.gradle.api.Project;
+import org.gradle.api.Task;
 import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.tasks.testing.Test;
 
+@SuppressWarnings("Convert2Lambda") // Gradle complains when using lambdas (build-cache won't wonk)
 public class QuarkusAppPlugin implements Plugin<Project> {
 
+  static final String START_TASK_NAME = "nessie-quarkus-start";
+  static final String STOP_TASK_NAME = "nessie-quarkus-stop";
+  static final String EXTENSION_NAME = "nessieQuarkusApp";
+  static final String CONFIG_NAME = "nessieQuarkusRunner";
+
   @Override
   public void apply(Project target) {
-    QuarkusAppExtension extension = target.getExtensions().create("quarkusAppRunnerProperties", QuarkusAppExtension.class, target);
+    QuarkusAppExtension extension = target.getExtensions().create(EXTENSION_NAME, QuarkusAppExtension.class, target);
 
-    final Configuration config = target.getConfigurations().create("quarkusAppRunnerConfig")
-      .setVisible(false)
-      .setDescription("The config for the Quarkus Runner.");
+    Configuration config = target.getConfigurations().create(CONFIG_NAME).setDescription("The config for the Nessie-Quarkus Runner.");
 
-    target.getTasks().register("quarkus-start", StartTask.class, new Action<StartTask>() {
+    // Cannot use the task name "test" here, because the "test" task might not have been registered yet.
+    // This `withType(Test.class...)` construct will configure any current and future task of type `Test`.
```

Comment:
current and future for the module or the whole project?

Reply:
It's the initialization order of plugins (the Gradle java-(base-)plugin isn't guaranteed to be configured when our plugin configures).

Reply:
does this mean quarkus-start binds to all modules in the project? Sorry...this is my gradle naivete showing

Reply:
Don't know what you're referring to ;)

Gradle plugins add their tasks - and the order of when plugins are initialized isn't "that deterministic".

Reply:
IIRC There are ways to add "plugin A depends on plugin B", but I ran out of time for the release. :(